internal/provider/resource_project_cloud_account_link.go

```diff
@@ -0,0 +1,668 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"wiz.io/hashicorp/terraform-provider-wiz/internal"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/client"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/utils"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/wiz"
+)
+
+// CloudAccountSearchResponse represents the response from a cloud account search.
+// It includes a GraphSearch object that contains a list of Nodes, each with a list of Entities.
+type CloudAccountSearchResponse struct {
+	GraphSearch struct {
+		Nodes []struct {
+			Entities []struct {
+				ID string `json:"id"`
+			} `json:"entities"`
+		} `json:"nodes"`
+	} `json:"graphSearch"`
+}
+
+// SearchForCloudAccountVars represents the variables for a cloud account search.
+// It includes the external ID, project ID, and a quick search flag.
+type SearchForCloudAccountVars struct {
+	ExternalID string `json:"externalId"`
+	ProjectID  string `json:"projectId"`
+	Quick      bool   `json:"quick"`
+}
+
+// PartialProjectWithCloudAccountLinks represents a partial project with cloud account links.
+// It includes a Project object.
+type PartialProjectWithCloudAccountLinks struct {
+	Project PartialProject `json:"project"`
+}
+
+// PartialProject represents a partial project.
+// It includes a list of cloud account links.
+type PartialProject struct {
+	CloudAccountLinks []*wiz.ProjectCloudAccountLink
+}
+
+// UpdateProjectCloudAccountLinks represents the input for updating project cloud account links.
+// It includes the ID and a Patch object.
+type UpdateProjectCloudAccountLinks struct {
```

Comment:
The correct type should be in internal/wiz/structs.go as `UpdateProjectInput`

Reply:
In order for that type to work I would have to add `json:"omitempty"` to all fields that currently don't have it. I don't know if that would break any existing resources.

Reply:
I suspect it would be breaking indeed. Then it's ok, just a comment to reflect this in the type similar to existing deviations would be great.